src/main/java/org/springframework/data/mapping/model/ClassGeneratingEntityInstantiator.java

```diff
@@ -64,11 +67,21 @@
 
 	private volatile Map<TypeInformation<?>, EntityInstantiator> entityInstantiators = new HashMap<>(32);
 
+	private final boolean fallbackToReflectionOnError;
```

Comment:
Introducing this flag seems to be related but not strictly required, is it?

Reply:
This is for testing purposes only. Tests were green during development although class-generated entity creation was broken and the creation fell back to reflective instance creation.